python/tvm/script/tir/sparse.py

```diff
@@ -0,0 +1,179 @@
+import synr
+import tvm
+from synr import ast
+from tvm.ir.base import Span
+from tvm.ir.expr import PrimExpr, Range
+
+from tvm.script.tir.node import BufferSlice
+from tvm.script.tir.utils import buffer_slice_to_region
+from .scope_handler import ScopeHandler, LoopInfo
+from .intrin import Intrin
+from ..context_maintainer import BlockInfo, ContextMaintainer
+from .special_stmt import SpecialStmt
+from tvm.tir.sparse import Axis, AxisTree, DenseFixedAxis, DenseVariableAxis, SpIterVar, SparseFixedAxis, SparseVariableAxis
+from typing import List, Mapping, Optional, Tuple, Any
+from tvm.runtime.object import Object
+from tvm.script.registry import register
+from ..utils import (
+    tvm_span_from_synr,
+    call_with_error_reporting,
+)
+
+
+@register
+class DenseFixed(SpecialStmt):
+    """Special Stmt for creating dense fixed axis.
+    """
+
+    def __init__(self):
+        def dense_fixed(
+            name: str,
+            length: PrimExpr,
+            idtype: str = 'int32'
+        ):
+            var_name = self.node.lhs[0].id.name
+            axis = DenseFixedAxis(name, length, idtype=idtype)
+            self.context.update_symbol(var_name, axis, self.node)
+        super().__init__(dense_fixed, def_symbol=True)
+
+
+@register
+class DenseVariable(SpecialStmt):
+    """Special Stmt for creating dense variable axis.
+    """
+
+    def __init__(self):
+        def dense_variable(
+            name: str,
+            length: PrimExpr,
+            indptr: tvm.tir.Var,
+            idtype: str = 'int32'
+        ):
+            var_name = self.node.lhs[0].id.name
+            axis = DenseVariableAxis(name, length, indptr, idtype=idtype)
+            self.context.update_symbol(var_name, axis, self.node)
+        super().__init__(dense_variable, def_symbol=True)
+
+
+@register
+class SparseFixed(SpecialStmt):
+    """Special Stmt for creating sparse fixed axis.
+    """
+
+    def __init__(self):
+        def sparse_fixed(
+            name: str,
+            shape: Tuple[PrimExpr, PrimExpr],
+            indices: tvm.tir.Var,
+            idtype: str = 'int32'
+        ):
+            var_name = self.node.lhs[0].id.name
+            length, num_cols = shape
+            axis = SparseFixedAxis(name, length, indices, num_cols, idtype=idtype)
+            self.context.update_symbol(var_name, axis, self.node)
+        super().__init__(sparse_fixed, def_symbol=True)
+
+
+@register
+class SparseVariable(SpecialStmt):
+    """Special Stmt for creating sparse variable axis:
+    """
+
+    def __init__(self):
+        def sparse_variable(
+            name: str,
+            shape: Tuple[PrimExpr, PrimExpr],
+            data: Tuple[tvm.tir.Var, tvm.tir.Var],
+            idtype: str = 'int32'
+        ):
+            var_name = self.node.lhs[0].id.name
+            length, nnz = shape
+            indptr, indices = data
+            axis = SparseVariableAxis(name, length, nnz, indptr, indices, idtype=idtype)
```

Comment:
I took a look at the demo carefully. Looks like `indptr` and `indices` here are of type `tir.handle`, so shall we use `buffer_decl` here as well to create the buffers?

Reply:
yes, it's missing. Fix soon.